### ISSUE TITLE
ddg.init crash on windows

### ISSUE THREAD

**Issue by tfjmp** [End-to-end-provenance/RDataTracker#116]
On ddginit

29 Jun 2016 18:35:18 [rdesktop] ERROR system error 2 (The system cannot find the file specified); OCCURRED AT: virtual void rstudio::core::http::NamedPipeAsyncClient::connectAndWriteRequest() C:/Users/Administrator/rstudio/src/cpp/core/include/core/http/NamedPipeAsyncClient.hpp:84; LOGGED FROM: void rstudio::desktop::NetworkReply::onError(const rstudio::core::Error&) C:\Users\Administrator\rstudio\src\cpp\desktop\DesktopNetworkReply.cpp:288
29 Jun 2016 18:35:28 [rdesktop] ERROR system error 2 (The system cannot find the file specified); OCCURRED AT: virtual void rstudio::core::http::NamedPipeAsyncClient::connectAndWriteRequest() C:/Users/Administrator/rstudio/src/cpp/core/include/core/http/NamedPipeAsyncClient.hpp:84; LOGGED FROM: void rstudio::desktop::NetworkReply::onError(const rstudio::core::Error&) C:\Users\Administrator\rstudio\src\cpp\desktop\DesktopNetworkReply.cpp:288
29 Jun 2016 18:35:38 [rdesktop] ERROR system error 2 (The system cannot find the file specified); OCCURRED AT: virtual void rstudio::core::http::NamedPipeAsyncClient::connectAndWriteRequest() C:/Users/Administrator/rstudio/src/cpp/core/include/core/http/NamedPipeAsyncClient.hpp:84; LOGGED FROM: void rstudio::desktop::NetworkReply::onError(const rstudio::core::Error&) C:\Users\Administrator\rstudio\src\cpp\desktop\DesktopNetworkReply.cpp:288

Would that ring some bell to anyone?

**Comment by tfjmp** [comment]
Error seem linked to RStudo itself rather than ddg.init().

Is there a way to set RStudio to be verbose and log literally everything in some files? It is quite terrible in term of feedback. Logs files (rdesktop.log and rsession-XXX.log) are empty at the exception of the above in rdesktop.log.

**Comment by tfjmp** [comment]
Ok playing with the debbuger inside ddg.init().

It seems to crash at:
ddg.write.timestamp.to.history()

.ddghistory file at that point is still empty.

**Comment by tfjmp** [comment]
ddg.history.file and ddg.path variable seems correct. However, ddgdir is NULL.

**Comment by tfjmp** [comment]
Crash in:
timestamp(quiet = TRUE)
-> .External2(C_addhistory, stamp)

**Comment by tfjmp** [comment]
It seems Barbara run though that issue before?
http://stackoverflow.com/questions/22566317/getanywheretimestamp-finds-two-functions-from-within-rstudio

**Comment by erboose** [comment]
Note that this version of RDataTracker works fine in RStudio on Windows when RDataTracker.R is sourced.  The problem occurs when RDataTracker is loaded as an installed package.

**Comment by blernermhc** [comment]
In the email thread that was started before this issue was opened, there was mention of needing to install the magrittr package.  I don't think we use that.  It was something Moe was looking at when trying to set up a pipe between R and Java.  I would try removing that and a) seeing if RDataTracker runs without it, and b) whether that helps with the install issue.

**Comment by blernermhc** [comment]
Also, the Windows timestamp problem is issue #7 in github, so, yes, we have run into that but not as something that crashed R.

Question:  is the problem when trying to install RDataTracker or when trying to use it after it has been installed.  I thought it was an installation problem, but the error message at the top is referring to ddginit, which sounds like a function that would be executed when RDataTracker is being run.  Which is it?

**Comment by tfjmp** [comment]
Installation goes fine. Problem is RStudio with dev branchement crash on ddg.init, for me.

It has dependencies to magrittr on the dev branch. But that's a spare issue #115.

**Comment by erboose** [comment]
By process of elimination, the problem appears to result from the following statement in the NAMESPACE file:

importFrom(utils, capture.output, head, object.size, packageVersion, savehistory, tail, timestamp, write.csv, write.table)

When this statement is commented out, R no longer crashes and the breakpoint and Rmarkdown features seem to work correctly.  Need to check if this causes any other problems.

**Comment by tfjmp** [comment]
Let me know if that fixed the issue:
https://github.com/End-to-end-provenance/RDataTracker/tree/windows_fix

**Comment by erboose** [comment]
Apparently utils and grDevices are part of the R system library and don't need to be imported.  If these are removed from NAMESPACE, the resulting install file works with RStudio / Windows and passes the regression tests.

**Comment by tfjmp** [comment]
#117 merged closing.

**Comment by blernermhc** [comment]
Right, when Alex ran Cran check, it came up with a lot of errors.  Apparently, the cran rules recently changed and they now require an explicit import of items from packages that are loaded by default.  Here is the stackoverflow answer that led us to this solution.

http://stackoverflow.com/questions/31311656/r-check-with-r-devel-gives-global-function-notes-related-to-core-package-functio

My suggestions for now are that we live with the cran check errors.  That is less bad than having a package that doesn't work when loaded.  We should also try to re-create a small example and send a bug report to RStudio.

**Comment by tfjmp** [comment]
It actually worked with just timestamp removed from thé import list. Would it pass CRAN test with everything but thé timestamp import?

**Comment by erboose** [comment]
Alas removing just timestamp did not work on Windows.

**Comment by tfjmp** [comment]
When removing the timestamp only, the timestamp is actually not saved and we got an error on ddg.save(), correct?

**Comment by tfjmp** [comment]
What I have here:
only removing timestamp: ddg.init() does not crash anymore, but get an error in ddg.save() [elsewhere too?]
removing everything seems to work.

The IMPORT in NAMESPACE seems to have various misunderstood side effect on RSTUDIO/Windows.

**Comment by erboose** [comment]
No crash but the following warnings are returned when running the install package on RStudio / Windows with only timestamp removed:

Warning messages:
1: In .ddg.insert.error.message(error.msg) :
  Part of history is missing. DDG may be incomplete! Tried reading from ./test_ddg/.ddghistory but could not find timestamp: ##------ Thu Jun 30 11:24:38 2016 ------##
2: In .ddg.insert.error.message(error.msg) :
  Part of history is missing. DDG may be incomplete! Tried reading from ./test_ddg/.ddghistory but could not find timestamp: ##------ Thu Jun 30 11:24:38 2016 ------##
3: In if (as.numeric(snum) > 0) line <- paste(snum, ":", lnum, sep = "") else line <- lnum :
  the condition has length > 1 and only the first element will be used

**Comment by MKLau** [comment]
Quick discussion with Barb and Emery. This is likely an issue with the importing of functions in RStudio on Windows.

**Comment by blernermhc** [comment]
There is no longer a crash but the history file does not seem to get written to on Windows.  See issue #431
